### ISSUE TITLE
E2E testing

### ISSUE THREAD

**Issue by oren-zohar** [elastic/cloudbeat#832]
**Summary**
Before continuing with the rest of the priority zero tasks, we need to verify our application works end-to-end to discover any critical issues

**Definition of done**
- [x] our solution works OOTB with the latest image, with the integration, using the cloud formation script deployment
- [x] our solution works OOTB with the latest image, with the integration, locally
- [x] scanning images completely (not encrypted)
- [x] review logs for any unexpected errors

> **Note**
> The verification should include multiple cycles and images

**Comment by jeniawhite** [comment]
Woking on the E2E, I've opened the following tickets:
- https://github.com/elastic/kibana/issues/154233
- https://github.com/elastic/cloudbeat/issues/852
- https://github.com/elastic/cloudbeat/issues/853
- https://github.com/elastic/cloudbeat/issues/854

**Comment by tehilashn** [comment]
@jeniawhite - thank you for the thorough work you've been doing in this e2e testing!

As part of this task, I recommend we also verify that we did not create any regression for KSPM and CSPM. 
Can you we please test the following? 

- [ ] Test # 1: install CSPM and KSPM using the 8.8 agent and cloudbeat (you can add this to your existing environment) 
- [ ] Test # 2: 
(1) Set up 8.7 ELK with CSPM and KSPM Integrations sending data
(2) Upgrade ELK to 8.8, ensure 8.7 version agent keeps working
(3) Deploy 8.8 version agent, and ensure it works as expected

**Comment by olegsu** [comment]
More issues that were found by @jeniawhite 

Kibana
- [ ] Missing fields and empty rows in the finding
- [ ] Transform health check posture type is incorrect, we should verify it in a different way
- [ ] The transform seems to cut off a lot of findings


Cloudbeat
- [ ] Add the missing permission for the `commonDataProvider` (related to https://github.com/elastic/cloudbeat/pull/864, https://github.com/elastic/security-team/issues/6339)
- [ ] Document the dev flow to support ssh connection to the instance and the snapshot link to download the cloudbeat



**Empty rows**
![Image](https://user-images.githubusercontent.com/23185631/232719813-5b08e78f-227c-4798-8f9e-bd009ffa50f3.png)

**Comment by jeniawhite** [comment]
Attempted to do the upgrade flow for 8.7.1 -> 8.8 BC.
Doing that I've deployed KSPM (on kind) and CSPM (on EC2 as binary).
I've noticed that the fleet doesn't show the upgrade button for KSPM, assumed due to this deployment being on top of Kubernetes which would require the fleet to change the manifest's image value.
I saw that there was an upgrade option for the CSPM and I clicked it, but besides the UI showing that the agent is upgrading, it did not recover.
I assumed that this is due to the fact that I was attempting to upgrade to a BC version.
I've verified that both agents kept on sending data and performing cycles after the stack upgrade with still running 8.7.1 versions.
In order to continue I've redeployed the binary with the latest version (CSPM on EC2), and changed the image manually in the manifest of the KSPM.
Continued to attempt and deploy vulnerability via the FE using the cloud formation button, this also did not work due to this being a BC version and the build not being available in the artifacts.
Resorted to deploying the cloud formation manually using the AWS CLI and taking the manifest from the repository.
Due to the above, I'm not sure how I'm supposed to check the upgrade without doing any hacks (changing images and reinstalling binaries).
